File: app/templates/static/scripts/calculadora.js
const formulario = document.getElementById('formulario');
const options = { style: 'currency', currency: 'EUR' };
const numberFormat = new Intl.NumberFormat('es-ES', options);

formulario.addEventListener('submit', function(event){
    event.preventDefault();

    const data = new FormData(formulario);

    console.log([...data])
  
    console.log(document.getElementById('ahorros').value)
    console.log(typeof(document.getElementById('ahorros').value))
    s = {
        "ahorros": parseFloat(document.getElementById('ahorros').value),
        "capitalAportado": parseFloat(document.getElementById("capitalAportado").value),
        "capitalInmueble": parseFloat(document.getElementById('capitalInmueble').value),
        "cuota": 0.0,
        "esPrimeraVivienda": true,
        "fechaNacimiento": "hoy",
        "nomina": parseFloat(document.getElementById('nomina').value),
        "otrosPrestamos": parseFloat(document.getElementById('otrosPrestamos').value),
        "plazo": parseInt(document.getElementById('plazo').value),
        "plazoRestante": 0,
        "prestamo": 0.0,
        "tasaInteres": 0.0,
        "tipoInteres": "fijo",
        "totalIntereses": 0.0
    }

    fetch('http://127.0.0.1:8859/api/calculadora',{
        method: "POST",
        body: JSON.stringify(s),
        headers: {
            "Content-Type": "application/json",
            "Authorization" : "Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VybmFtZSI6Ik1hcmNvcyIsInBhc3N3b3JkIjoxMjMsImV4cCI6MTY1Nzc5NjIyNH0.tEATWj83zVpwUduR4o96CEortX-jgkCAxIZ65KLZOg8"
        }
    })

    .then(res => res.json())
    .then(data => {
        console.log(data.porcentaje)
        imprimirTabla(data.hipoteca.amortizaciones);
        imprimirTarjetas(data)
    
    })
    .catch(err => console.log(err))

})

//Método para pintar datos en pantalla
function imprimirTabla(datos){
    var html = ''
    for (let i = 0; i < datos.length; i++) {
        var row = `<tr>
                <td>${datos[i].numeroCuota}</td>
                <td>${numberFormat.format(datos[i].cuota)}</td>
                <td>${numberFormat.format(datos[i].interes)}</td>
                <td>${numberFormat.format(datos[i].cuotaAmortizacion)}</td>
                <td>${numberFormat.format(datos[i].totalAmortizacion)}</td>
                <td>${numberFormat.format(Math.abs(datos[i].capitalPorAmortizar))}</td>
              </tr>`

        html = html + row
    }

    document.querySelector('#table_id > tbody').innerHTML = html;

}


//Método para pintar datos en pantalla
function imprimirTarjetas(datos){

    document.getElementById('precioInmuebleCard').innerHTML = numberFormat.format(datos.hipoteca.capitalInmueble);
    document.getElementById('importeInicialCard').innerHTML = numberFormat.format(datos.hipoteca.capitalAportado);
    document.getElementById('prestamoCard').innerHTML = numberFormat.format(datos.hipoteca.prestamo);
    document.getElementById('cuotaMensualCard').innerHTML = numberFormat.format(datos.hipoteca.cuota);
    document.getElementById('tipointeresCard').innerHTML = datos.hipoteca.tipoInteres;
    document.getElementById('interesCard').innerHTML = Number.parseFloat(datos.hipoteca.tasaInteres).toFixed(2) + '%';
    document.getElementById('totalInteresCard').innerHTML = numberFormat.format(datos.hipoteca.totalIntereses);
    document.getElementById('porcentajeCard').innerHTML = Number.parseFloat(datos.porcentaje).toFixed(2) + '%';
    

}

File: app/routes/calculadora.py
import json
from types import SimpleNamespace

from flask import Blueprint , request , Flask
from flask_cors import cross_origin , CORS

from app.function_jwt import validarToken

from app.models.Hipoteca import Hipoteca
from app.models.HipotecaDTO import HipotecaDTO
from app.models.Simulacion import Simulacion
from app.services.impl.HipotecaFijaServicio import HipotecaFijaServicio
from app.services.impl.HipotecaVariableServicio import HipotecaVariableServicio
from app.services.impl.SimulacionServicio import SimulacionServicio

calculadora_blueprint = Blueprint("calculadora_blueprint", __name__)

hipotecaFijaServicio = HipotecaFijaServicio()
hipotecaVariableServicio = HipotecaVariableServicio()
simulacionServicio = SimulacionServicio()


@calculadora_blueprint.before_request
def validar_token_middleware():
    print(1)
    token = request.headers['Authorization'].split(" ")[1]
    return validarToken(token,False)

@calculadora_blueprint.route("/calculadora", methods=["POST"])
def calculadora():

    print(request.headers['Authorization'].split(" ")[1])
    # Almacenar el JSON recibido en un objeto Hipoteca
    print(type(request.json['capitalInmueble']))
    hipoteca = Hipoteca(request.json['totalIntereses'], request.json['capitalInmueble'],
                        request.json["capitalAportado"] , request.json["prestamo"] , request.json["cuota"],
                        request.json["plazo"] , request.json["plazoRestante"] , request.json["tasaInteres"],
                        request.json["tipoInteres"] , request.json["fechaNacimiento"] , request.json["ahorros"],
                        request.json["nomina"],request.json["otrosPrestamos"], request.json["esPrimeraVivienda"] )


    simulacion = Simulacion(None,None,None)
    simulacionServicio.generarHipoteca(hipoteca,simulacion)
    porcentaje = simulacionServicio.calcularProbabilidad(simulacion)

    hipoteca.amortizaciones = list()

    if request.json["tipoInteres"] == "fijo":
        hipotecaFijaServicio.calcularAmortizaciones(hipoteca)
        print("cuota" + str(hipoteca.cuota))
    else:
        hipotecaVariableServicio.calcularAmortizaciones(hipoteca)

    response = HipotecaDTO(hipoteca,porcentaje)

    return response.toJson()